### ISSUE TITLE
Error waiting for a debug connection: The log reader stopped unexpectedly. Error launching application on Windows.

### ISSUE THREAD

**Issue by OcaenEyes** [flutter/flutter#93635]
1. my application is create by `flutter create  xxx`,  when i  use  `flutter run -d windows`  in cmd

2. but it doesn't work, and throw error log:
```
Launching lib\main.dart on Windows in debug mode...
Building Windows application...                                         
Error waiting for a debug connection: The log reader stopped unexpectedly.
Error launching application on Windows.
```

3. then i use `flutter doctor -v` ,the result log:

```
[√] Flutter (Channel stable, 2.5.3, on Microsoft Windows [Version 10.0.22000.318], locale zh-CN)
    • Flutter version 2.5.3 at C:\Developer\flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 18116933e7 (4 weeks ago), 2021-10-15 10:46:35 -0700
    • Engine revision d3ea636dc5
    • Dart version 2.14.4
    • Pub download mirror https://pub.flutter-io.cn
    • Flutter download mirror https://storage.flutter-io.cn

[√] Android toolchain - develop for Android devices (Android SDK version 31.0.0)
    • Android SDK at C:\Developer\AndroidSdk
    • Platform android-31, build-tools 31.0.0
    • ANDROID_HOME = C:\Developer\AndroidSdk
    • Java binary at: C:\Program Files\Android\Android Studio\jre\bin\java
    • Java version OpenJDK Runtime Environment (build 11.0.10+0-b96-7249189)
    • All Android licenses accepted.

[√] Chrome - develop for the web
    • Chrome at C:\Program Files\Google\Chrome\Application\chrome.exe

[√] Visual Studio - develop for Windows (Visual Studio Community 2019 16.11.6)
    • Visual Studio at C:\Program Files (x86)\Microsoft Visual Studio\2019\Community
    • Visual Studio Community 2019 version 16.11.31829.152
    • Windows 10 SDK version 10.0.19041.0

[√] Android Studio (version 2020.3)
    • Android Studio at C:\Program Files\Android\Android Studio
    • Flutter plugin can be installed from:
       https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
       https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 11.0.10+0-b96-7249189)

[√] IntelliJ IDEA Ultimate Edition (version 2021.1)
    • IntelliJ at C:\Program Files\JetBrains\IntelliJ IDEA 2021.1.3
    • Flutter plugin can be installed from:
       https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
       https://plugins.jetbrains.com/plugin/6351-dart

[√] Connected device (3 available)
    • Windows (desktop) • windows • windows-x64    • Microsoft Windows [Version 10.0.22000.318]
    • Chrome (web)      • chrome  • web-javascript • Google Chrome 95.0.4638.69
    • Edge (web)        • edge    • web-javascript • Microsoft Edge 95.0.1020.44

• No issues found!
```

**Comment by cuishijie1991** [comment]
Same issues! 
Same flutter and Visual Studio version
mark+1

**Comment by maheshmnj** [comment]
Hi @OcaenEyes, thanks for filing the issue. I could not reproduce it, Can you please try running `flutter clean` and see if that makes any difference, Also does this issue occur with a specific project or a newly created flutter project too?

<details>
<summary>flutter doctor -v</summary>

```
[√] Flutter (Channel stable, 2.5.3, on Microsoft Windows [Version 10.0.19042.1348], locale en-US)
    • Flutter version 2.5.3 at C:\flutter_sdk\stable
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 18116933e7 (4 weeks ago), 2021-10-15 10:46:35 -0700
    • Engine revision d3ea636dc5
    • Dart version 2.14.4

[√] Android toolchain - develop for Android devices (Android SDK version 31.0.0)
    • Android SDK at C:\Users\mahesh\AppData\Local\Android\sdk
    • Platform android-31, build-tools 31.0.0
    • Java binary at: C:\Program Files\Android\Android Studio\jre\bin\java      
    • Java version OpenJDK Runtime Environment (build 11.0.10+0-b96-7249189)    
    • All Android licenses accepted.

[√] Chrome - develop for the web
    • Chrome at C:\Program Files\Google\Chrome\Application\chrome.exe

[√] Visual Studio - develop for Windows (Visual Studio Community 2019 16.11.4)      
    • Visual Studio at C:\Program Files (x86)\Microsoft Visual Studio\2019\Community
    • Visual Studio Community 2019 version 16.11.31727.386
    • Windows 10 SDK version 10.0.19041.0

[√] Android Studio (version 2020.3)
    • Android Studio at C:\Program Files\Android\Android Studio
    • Flutter plugin can be installed from:
       https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
       https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 11.0.10+0-b96-7249189)        

[√] VS Code (version 1.61.2)
    • VS Code at C:\Users\mahesh\AppData\Local\Programs\Microsoft VS Code
    • Flutter extension version 3.28.0

[√] Connected device (3 available)
    • Windows (desktop) • windows • windows-x64    • Microsoft Windows [Version 10.0.19042.1348]
    • Chrome (web)      • chrome  • web-javascript • Google Chrome 95.0.4638.69
    • Edge (web)        • edge    • web-javascript • Microsoft Edge 95.0.1020.44

• No issues found!
```
</details>

**Comment by OcaenEyes** [comment]
Hi@maheshmnj，thank you for help . i have tried `flutter clean`,and the rusult log：
```Deleting build...                                                   72ms
Deleting .dart_tool...                                              12ms
Deleting .packages...                                                0ms
Deleting Generated.xcconfig...                                       0ms
Deleting flutter_export_environment.sh...                            0ms
Deleting ephemeral...                                                0ms
Deleting ephemeral...                                                1ms
Deleting ephemeral...                                               31ms
```

then i use `flutter run -d windows -v`,but it still didn‘t work，the error log is :
```
[  +13 ms] FinalizeBuildStatus:
[        ]   正在删除文件“x64\Debug\INSTALL\INSTALL.tlog\unsuccessfulbuild”。
[        ]   正在对“x64\Debug\INSTALL\INSTALL.tlog\INSTALL.lastbuildstate”执行 Touch 任务。
[        ] 已完成生成项目“E:\GitHub\FlutterLN\oceanbbs\build\windows\INSTALL.vcxproj”(默认目标)的操作。        
[  +13 ms] 已成功生成。
[        ]     0 个警告
[        ]     0 个错误
[        ] 已用时间 00:00:23.45
[  +17 ms] Building Windows application... (completed in 31.0s)
[ +717 ms] Error waiting for a debug connection: The log reader stopped unexpectedly.
[   +2 ms] Error launching application on Windows.
[   +1 ms] "flutter run" took 32,927ms.
[   +3 ms]
           #0      throwToolExit (package:flutter_tools/src/base/common.dart:10:3)
           #1      RunCommand.runCommand (package:flutter_tools/src/commands/run.dart:669:9)
           <asynchronous suspension>
           #2      FlutterCommand.run.<anonymous closure>
(package:flutter_tools/src/runner/flutter_command.dart:1125:27)
           <asynchronous suspension>
           #3      AppContext.run.<anonymous closure> (package:flutter_tools/src/base/context.dart:150:19)     
           <asynchronous suspension>
           #4      CommandRunner.runCommand (package:args/command_runner.dart:209:13)
           <asynchronous suspension>
           #5      FlutterCommandRunner.runCommand.<anonymous closure>
           (package:flutter_tools/src/runner/flutter_command_runner.dart:288:9)
           <asynchronous suspension>
           #6      AppContext.run.<anonymous closure> (package:flutter_tools/src/base/context.dart:150:19)     
           <asynchronous suspension>
           #7      FlutterCommandRunner.runCommand
(package:flutter_tools/src/runner/flutter_command_runner.dart:236:5)
           <asynchronous suspension>
           #8      run.<anonymous closure>.<anonymous closure> (package:flutter_tools/runner.dart:62:9)        
           <asynchronous suspension>
           #9      AppContext.run.<anonymous closure> (package:flutter_tools/src/base/context.dart:150:19)     
           <asynchronous suspension>
           #10     main (package:flutter_tools/executable.dart:92:3)
           <asynchronous suspension>


[ +202 ms] ensureAnalyticsSent: 197ms
[   +1 ms] Running shutdown hooks
[        ] Shutdown hooks complete
[        ] exiting with code 1
````

on the other hand, all my  project were created by `flutter create xxx-appname-xx` , they ran success on my macos  when i used `flutter run -d macos`,
 but they throwed error message  `Error waiting for a debug connection: The log reader stopped unexpectedly. Error launching application on Windows` on my windows11 when i used `flutter run -d windows`,

**Comment by OcaenEyes** [comment]
i solved this problem.  @maheshmnj  @cuishijie1991
because  the sdk version .   choose  `win 11  sdk(10.0.22000.0)` in  visual studio installer
**windows 11  need  choose  `win 11  sdk(10.0.22000.0)`**
[](url)

**Comment by KnucklesEQ** [comment]
Thanks @OcaenEyes !!

I had the same problem today. Trying to mount a new Flutter environment on Windows, same error as yours: "_Error waiting for a debug connection: The log reader stopped unexpectedly. Error launching application on Windows._"

The error only occurred when booting in debug mode. In release or profile mode, there were no problems.

And the error only happens on Windows, no problem to execute the app on Web.

The solution, as you suggested: I was previously using version `"10.0.19041.0"`, and when I installed `"10.0.22000.0"`, everything worked without problems. 

This was the first time I have faced a problem like this.

PS - I'm using Windows 10 (not Windows 11). And I've also installed Visual Studio Community 2019 (not 2022 Ed.)
